### PR TITLE
Add support for running with WASM

### PR DESCRIPTION
**Pull request by EricApostal**
This PR removes the need for `dart:html` and `package:js/js.dart`, replacing them with `package:web/web.dart` and `dart:js_interop` / `dart:js_interop_unsafe`. 

There are a few things untested here, particularly `firstTrack.oncuechange`, and I don't know enough about HLS to ensure it's properly supported. In testing, everything appears to function okay, running with --wasm or otherwise.

The "seamless" test yields a black screen, but it's my understanding that it's an issue with the test, as when I remove the `setAudioTrack` call it works fine (`setAudioTrack` throws an unsupported exception on web).

### CODE REVIEW

**Comment by abdelaziz-mahdy** [comment]
this test has failed for web 
```
❌ test/src/models/media/media_test.dart: media-http-headers-propagate (failed)
  Expected: throws <Instance of 'UnsupportedError'>
    Actual: <Closure 'main__closure'>
     Which: returned Media:<Media(https://user-images.githubusercontent.com/28951144/229373709-603a7a89-2105-4e1b-a5a5-a6c[35](https://github.com/media-kit/media-kit/actions/runs/12248188251/job/34192335012?pr=1059#step:8:41)67c9a59.mp4, extras: null, httpHeaders: {x: y}, start: null, end: null)>
  
  org-dartlang-sdk:///lib/_internal/js_runtime/lib/js_helper.dart 1198:19   Object.wrapException
  org-dartlang-sdk:///lib/_internal/js_runtime/lib/js_helper.dart 1239:28   Object.throwExpression
  package:matcher                                                           Object.expect
  test/src/models/media/media_test.dart 218:7                               main.<fn>
  package:test_api                                                          <fn>
  org-dartlang-sdk:///lib/_internal/js_runtime/lib/async_patch.dart 303:19  _wrapJsFunctionForAsync.closure.$protected
  org-dartlang-sdk:///lib/_internal/js_runtime/lib/async_patch.dart 328:23  _wrapJsFunctionForAsync.<fn>
  org-dartlang-sdk:///lib/_internal/js_runtime/lib/async_patch.dart 279:19  _awaitOnObject.<fn>
```

**Comment by alexmercerind** [comment]
> this test has failed for web
> 
> ```
> ❌ test/src/models/media/media_test.dart: media-http-headers-propagate (failed)
>   Expected: throws <Instance of 'UnsupportedError'>
>     Actual: <Closure 'main__closure'>
>      Which: returned Media:<Media(https://user-images.githubusercontent.com/28951144/229373709-603a7a89-2105-4e1b-a5a5-a6c[35](https://github.com/media-kit/media-kit/actions/runs/12248188251/job/34192335012?pr=1059#step:8:41)67c9a59.mp4, extras: null, httpHeaders: {x: y}, start: null, end: null)>
>   
>   org-dartlang-sdk:///lib/_internal/js_runtime/lib/js_helper.dart 1198:19   Object.wrapException
>   org-dartlang-sdk:///lib/_internal/js_runtime/lib/js_helper.dart 1239:28   Object.throwExpression
>   package:matcher                                                           Object.expect
>   test/src/models/media/media_test.dart 218:7                               main.<fn>
>   package:test_api                                                          <fn>
>   org-dartlang-sdk:///lib/_internal/js_runtime/lib/async_patch.dart 303:19  _wrapJsFunctionForAsync.closure.$protected
>   org-dartlang-sdk:///lib/_internal/js_runtime/lib/async_patch.dart 328:23  _wrapJsFunctionForAsync.<fn>
>   org-dartlang-sdk:///lib/_internal/js_runtime/lib/async_patch.dart 279:19  _awaitOnObject.<fn>
> ```

Related to #1043.

**Comment by EricApostal** [comment]
@alexmercerind It seems like it was a test that should have been removed in the PR mentioned. I removed the web-specific test case and grouped it with the other, which now runs fine.

File: media_kit_video/lib/src/video_controller/web_video_controller/real.dart [hunk 1]
```diff
@@ -49,7 +50,11 @@ class WebVideoController extends PlatformVideoController {
     player.platform?.release.add(controller._dispose);
 
     // Retrieve the [html.VideoElement] instance from [js.context].
-    controller._element = js.context[_kInstances][handle];
+    var propertyGet = globalContext.getProperty(_kInstances.toJS);
+    // controller._element =
```
Comment (alexmercerind) [inline media_kit_video/lib/src/video_controller/web_video_controller/real.dart:54]: Remove redundant comment.

File: media_kit_video/lib/src/video_controller/web_video_controller/real.dart [hunk 2]
```diff
@@ -49,7 +50,11 @@ class WebVideoController extends PlatformVideoController {
     player.platform?.release.add(controller._dispose);
 
     // Retrieve the [html.VideoElement] instance from [js.context].
-    controller._element = js.context[_kInstances][handle];
+    var propertyGet = globalContext.getProperty(_kInstances.toJS);
```
Comment (alexmercerind) [inline media_kit_video/lib/src/video_controller/web_video_controller/real.dart:53]: Use `final` keyword. Let's call `propertyGet` simply `element`.

**Review by alexmercerind — APPROVED** [review]
Thanks a lot for your contribution. 

Few nits from my end, LGTM otherwise!

**Comment by alexmercerind** [comment]
> The "seamless" test yields a black screen, but it's my understanding that it's an issue with the test, as when I remove the setAudioTrack call it works fine (setAudioTrack throws an unsupported exception on web).

Can you add some checks to prevent calls to that on web?

File: media_kit_video/lib/src/video_controller/web_video_controller/real.dart (hunk 2 above)
Comment (EricApostal) [reply media_kit_video/lib/src/video_controller/web_video_controller/real.dart]: https://github.com/EricApostal/media-kit/blob/1749c03465c9e9be8d1446d1cf513ba6b5a5dc81/media_kit_video/lib/src/video_controller/web_video_controller/real.dart#L53 changed

File: media_kit_video/lib/src/video_controller/web_video_controller/real.dart (hunk 1 above)
Comment (EricApostal) [reply media_kit_video/lib/src/video_controller/web_video_controller/real.dart]: done https://github.com/EricApostal/media-kit/blob/1749c03465c9e9be8d1446d1cf513ba6b5a5dc81/media_kit_video/lib/src/video_controller/web_video_controller/real.dart#L53

File: media_kit_test/README.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+This project is a starting point for a Flutter application.
+
+A few resources to get you started if this is your first Flutter project:
+
+- [Lab: Write your first Flutter app](https://docs.flutter.dev/get-started/codelab)
+- [Cookbook: Useful Flutter samples](https://docs.flutter.dev/cookbook)
+
+For help getting started with Flutter development, view the
+[online documentation](https://docs.flutter.dev/), which offers tutorials,
+samples, guidance on mobile development, and a full API reference.
```
[the remark this answers is not in the document]
Comment (EricApostal) [reply media_kit_test/README.md]: Should be removed now

**Comment by EricApostal** [comment]
I added a runtime check for the seamless web check, which works fine when testing. 
https://github.com/EricApostal/media-kit/commit/1749c03465c9e9be8d1446d1cf513ba6b5a5dc81#diff-a94f5cde7c0e38f0b1eef3910676195c20865ed06ebd2fb81d8c38739f21aaadR59

**Comment by EricApostal** [comment]
Okay, I looked through and I everything looks good on my end. Let me know if y'all need any more changes done.

**Comment by alexmercerind** [comment]
@EricApostal 

Thanks again. My web knowledge is bit rusty, why am I seeing this in console? Is it just me?

image

**Comment by EricApostal** [comment]
> @EricApostal
> 
> Thanks again. My web knowledge is bit rusty, why am I seeing this in console? Is it just me?
> image


@alexmercerind that looks like a CORS response error. The URL endpoint sets a response header to disallow your browser from loading media from different domains, usually for security reasons. Ex: You can't call the discord api from your own website directly, because Discord sees that as a security risk, thus disables it. In what project are you seeing this error?

**Comment by EricApostal** [comment]
Wait, sorry, what message are you referring to?

**Comment by alexmercerind** [comment]
Yeah the same one... these videos used to play previously IIRC.

**Comment by EricApostal** [comment]
That's odd. Where can I test to reproduce this?

**Comment by EricApostal** [comment]
> That's odd. Where can I test to reproduce this?

Like in what usage of media kit does this occur? The test project does not yield this error for me.

image

**Comment by alexmercerind** [comment]
I assume something funny is happening only with me... the existing videos do not seem to load. Also, video is oddly outside the viewport. I am on the latest version (just downloaded) of Google Chrome.

Anyways... merging. We can deal with (if any) problems in later pull-requests. 

Thanks!

**Pull request merged**